<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RouteEditor</class>
 <widget class="QDialog" name="RouteEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1215</width>
    <height>740</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>740</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Route Editor</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,0">
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,1,0">
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>New Inbound</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QPushButton" name="addInboundBtn">
          <property name="toolTip">
           <string>Add Inbound</string>
          </property>
          <property name="text">
           <string>Editor</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="addDefaultBtn">
          <property name="toolTip">
           <string>Add Inbound from Global Settings</string>
          </property>
          <property name="text">
           <string>Global Settings</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>New Outbound</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>New Outbound</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="importOutboundBtn">
            <property name="toolTip">
             <string>Import Outbound</string>
            </property>
            <property name="text">
             <string>Import</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addOutboundBtn">
            <property name="toolTip">
             <string>Add Outbound</string>
            </property>
            <property name="text">
             <string>Editor</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="insertBlackBtn">
            <property name="toolTip">
             <string>Add Blackhole Outbound</string>
            </property>
            <property name="text">
             <string>Blackhole</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="insertDirectBtn">
            <property name="toolTip">
             <string>Add Freedom Outbound</string>
            </property>
            <property name="text">
             <string>Freedom</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addBalancerBtn">
            <property name="toolTip">
             <string>Add Balancer</string>
            </property>
            <property name="text">
             <string>Balancer</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addChainBtn">
            <property name="toolTip">
             <string>Add Proxy Chain</string>
            </property>
            <property name="text">
             <string>Chain</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1,1,0,0">
          <item>
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Existing Connection</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="importGroupBtn">
            <property name="toolTip">
             <string>Group</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="importConnBtn">
            <property name="toolTip">
             <string>Connection</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="importExistingBtn">
            <property name="toolTip">
             <string>Copy Connection</string>
            </property>
            <property name="text">
             <string>Copy</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="linkExistingBtn">
            <property name="toolTip">
             <string>Reference Connection</string>
            </property>
            <property name="text">
             <string>Reference</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="title">
        <string>Connection Template</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Inbound</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>Outbound</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>Routing Settings</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Domain Strategy</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="domainStrategyCombo">
          <item>
           <property name="text">
            <string notr="true">AsIs</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">IPIfNonMatch</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string notr="true">IPOnDemand</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>Default Outbound</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="defaultOutboundCombo">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QTabWidget" name="nodesTab">
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="tabWidget_3Page1">
      <attribute name="title">
       <string>Route Editor</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="1">
       <item>
        <widget class="QWidget" name="ruleEditorUIWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>150</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Chain Editor</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QWidget" name="chainEditorUIWidget" native="true"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>DNS Settings</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="1" column="0">
        <widget class="QWidget" name="dnsEditorUIWidget" native="true"/>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="checkBox">
         <property name="text">
          <string>Override global DNS settings</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Misc</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>Browser Forwarder</string>
         </property>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Listening Address</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="bfListenIPTxt"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>:Listening Port</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="bfListenPortTxt">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>8080</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_7">
         <property name="title">
          <string> Observatory</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Subject Selector</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="obSubjectSelectorTxt"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="1">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="horizontalGroupBox">
       <property name="title">
        <string>Inbound / Outbound Info</string>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Tag</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="tagLabel">
          <property name="text">
           <string>Unknown</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Protocol</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="protocolLabel">
          <property name="text">
           <string>Unknown</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Host</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="hostLabel">
          <property name="text">
           <string>Unknown</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_19">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="portLabel">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>Rules Ordering</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="1" column="0">
         <widget class="QListWidget" name="ruleListWidget">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="dragEnabled">
           <bool>true</bool>
          </property>
          <property name="dragDropMode">
           <enum>QAbstractItemView::InternalMove</enum>
          </property>
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="verticalScrollMode">
           <enum>QAbstractItemView::ScrollPerPixel</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Drag and drop to re-order the rules.</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="2" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0,0">
     <item>
      <widget class="QLabel" name="statusLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Status</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="debugPainterCB">
       <property name="text">
        <string>Debug Connection Painter</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>addDefaultBtn</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RouteEditor</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1115</x>
     <y>697</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RouteEditor</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1115</x>
     <y>697</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
